### PR TITLE
Add clipBehavior to Chips and update documents

### LINKED ISSUE #20746: `[border]` is not found  in chip.dart
In the public API doc of chip.dart, we referenced `[border]` but there's no such field.

### PR DESCRIPTION
**Pull request by liyuqian**
This fixes https://github.com/flutter/flutter/issues/20746

### CODE REVIEW

File: packages/flutter/lib/src/material/chip.dart [hunk 1]
```diff
@@ -540,7 +548,8 @@ class InputChip extends StatelessWidget
   /// The [onPressed] and [onSelected] callbacks must not both be specified at
   /// the same time.
   ///
-  /// The [label], [isEnabled] and [selected] arguments must not be null.
+  /// The [label], [isEnabled], [selected], and [clipBehavior] arguments must
+  /// not be null.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/chip.dart]: Looks like you need an `assert(clipBehavior != nll)`

File: packages/flutter/lib/src/material/chip.dart [hunk 2]
```diff
@@ -419,7 +422,7 @@ abstract class TappableChipAttributes {
 class Chip extends StatelessWidget implements ChipAttributes, DeletableChipAttributes {
   /// Creates a material design chip.
   ///
-  /// The [label] argument must not be null.
+  /// The [label] and [clipBehavior] argument must not be null.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/chip.dart]: argument => arguments

File: packages/flutter/lib/src/material/chip.dart [hunk 3]
```diff
@@ -388,8 +391,8 @@ abstract class TappableChipAttributes {
 /// Supplying a non-null [onDeleted] callback will cause the chip to include a
 /// button for deleting the chip.
 ///
-/// Requires one of its ancestors to be a [Material] widget. The [label],
-/// [deleteIcon], and [border] arguments must not be null.
+/// Requires one of its ancestors to be a [Material] widget. The [label]
+/// argument must not be null.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/chip.dart]: label and clipBehavior arguments must not be null

**Review by HansMuller — APPROVED** [review]
LGTM

I'm assuming that Chips typically do not require clipping.

File: packages/flutter/lib/src/material/chip.dart (hunk 3 above)
Comment (liyuqian) [reply packages/flutter/lib/src/material/chip.dart]: Done

File: packages/flutter/lib/src/material/chip.dart (hunk 2 above)
Comment (liyuqian) [reply packages/flutter/lib/src/material/chip.dart]: Done

File: packages/flutter/lib/src/material/chip.dart (hunk 1 above)
Comment (liyuqian) [reply packages/flutter/lib/src/material/chip.dart]: Done.

**Pull request merged**